File: ios/Kwift Template/com/lightningkite/kwifttemplate/api/APIMock.shared.swift
//Package: com.lightningkite.kwifttemplate.api
//Converted using Kwift2

import Foundation
import Kwift
import KwiftMaps



public class APIMock: APIInterface {
    
    
    
    //Start Companion
    static public var delayMs: Int64 = 150
    static public var lorem = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    //End Companion
    
    
    public func getExamplePosts(onResult: @escaping (Int32, Array<Post>?, String?) -> Void) -> Void {
        delay(APIMock.delayMs) { () in 
            onResult(200, [Post(title: "First", body: APIMock.lorem)], nil)
        }
    }
    
    public init() {
    }
}
 

File: android/src/main/java/com/lightningkite/khrysalistemplate/api/APIMock.shared.kt
package com.lightningkite.khrysalistemplate.api

import com.lightningkite.khrysalis.delay
import com.lightningkite.khrysalis.escaping
import com.lightningkite.khrysalistemplate.models.Post
import io.reactivex.Single
import java.util.concurrent.TimeUnit

class APIMock() : APIInterface {

    companion object {
        var delayMs: Long = 150
        val lorem =
            "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    }

    override fun getExamplePosts(): Single<List<Post>> {
        return Single.just(listOf(Post(title = "First", body = APIMock.lorem))).delay(1000L, TimeUnit.MILLISECONDS)
    }
}
